qiskit_optimization/runtime/vqe_program.py

```diff
@@ -0,0 +1,378 @@
+                (approximate) eigenstate of the minimum eigenvalue main result and their expectation
+                values returned. For instance in chemistry these can be dipole operators, total
+                particle count operators so we can get values for these at the ground state.
+
+        Returns:
+            MinimumEigensolverResult
+
+        Raises:
+            ValueError: If the backend has not yet been set.
+            ValueError: If the provider has not yet been set.
+            RuntimeError: If the job execution failed.
+
+        """
+        if self.backend is None:
+            raise ValueError("The backend has not been set.")
+
+        if self.provider is None:
+            raise ValueError("The provider has not been set.")
+
+        # try to convert the operators to a PauliSumOp, if it isn't already one
+        operator = _convert_to_paulisumop(operator)
+        if aux_operators is not None:
+            aux_operators = [_convert_to_paulisumop(aux_op) for aux_op in aux_operators]
+
+        # combine the settings with the given operator to runtime inputs
+        inputs = {
+            "operator": operator,
+            "aux_operators": aux_operators,
+            "ansatz": self.ansatz,
+            "optimizer": self.optimizer,
+            "initial_point": self.initial_point,
+            "shots": self.shots,
+            "measurement_error_mitigation": self.measurement_error_mitigation,
+            "store_intermediate": self.store_intermediate,
+        }
+
+        # define runtime options
+        options = {"backend_name": self.backend.name()}
+
+        # send job to runtime and return result
+        job = self.provider.runtime.run(
+            program_id=self.program_id,
+            inputs=inputs,
+            options=options,
+            callback=self._wrap_vqe_callback(),
+        )
+        # print job ID if something goes wrong
+        try:
+            result = job.result()
+        except Exception as exc:
+            raise RuntimeError(f"The job {job.job_id()} failed unexpectedly.") from exc
+
+        # re-build result from serialized return value
+        vqe_result = VQEProgramResult()
+        vqe_result.job_id = job.job_id()
+        vqe_result.cost_function_evals = result.get("cost_function_evals", None)
+        vqe_result.eigenstate = result.get("eigenstate", None)
+        vqe_result.eigenvalue = result.get("eigenvalue", None)
+        vqe_result.aux_operator_eigenvalues = result.get("aux_operator_eigenvalues", None)
+        vqe_result.optimal_parameters = result.get("optimal_parameters", None)
+        vqe_result.optimal_point = result.get("optimal_point", None)
+        vqe_result.optimal_value = result.get("optimal_value", None)
+        vqe_result.optimizer_evals = result.get("optimizer_evals", None)
+        vqe_result.optimizer_time = result.get("optimizer_time", None)
+        vqe_result.optimizer_history = result.get("optimizer_history", None)
```

Comment:
Maybe extract it to a separate method?

Reply:
As this code is a duplicate of the one in Nature, from my viewpoint its best to continue to keep the code identical, so that hopefully in the future, it can go in one place and then we don't have to reconcile differences between them. Hence if this is done here I think it should be done in Nature. Since we are nearing a release with everyone busy my reaction would be to let it be as-is at present.

Reply:
I already changed this locally, I don't mind either way. Since it's already done I'll push it now but we can also revert.

Reply:
We might keep the version above for the sake of consistency as Steve suggests.